hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -518,11 +541,26 @@ private Map<String, String> convertToMetadata(Database database) {
     if (database.getDescription() != null) {
       meta.put("comment", database.getDescription());
     }
+    if (database.getOwnerName() != null) {
+      meta.put(HMS_DB_OWNER, database.getOwnerName());
+      if (database.getOwnerType() != null) {
+        meta.put(HMS_DB_OWNER_TYPE, database.getOwnerType().name());
+      }
+    }
 
     return meta;
   }
 
   Database convertToDatabase(Namespace namespace, Map<String, String> meta) {
+    Preconditions.checkArgument(
+        meta.get(HMS_DB_OWNER_TYPE) == null
```

Comment:
Optional: A bit related to previous comment, I actually think here there may not be such a need for this extra bulky check, and call to a third party library?  If I'm not mistaken the valueOf anyway throws IllegalArgumentException if it does not match a value, and it would be pretty easy for user to find out?  

not sure if it throws exception for nulls, if not we could do that I suppose. 

Just trying to reduce bulky code here if possible.

Reply:
Yes, Enum.Valueof will through IllegalArgumentException if string is invalid or null. I originally was following the principal of "if something has to fail then fail early". In this case, I would take your recommendation because this function is too small and it's not much "earlier" + yes, this error could be figured out by users

Reply:
Thanks for understanding!